core/src/executor/sort.rs

```diff
@@ -41,48 +48,100 @@ impl<'a> Sort<'a> {
 
     pub async fn apply(
         &self,
-        rows: impl Stream<Item = Result<AggregateContext<'a>>> + 'a,
-    ) -> Result<Pin<Box<dyn Stream<Item = Item<'a>> + 'a>>> {
+        rows: impl Stream<Item = Result<(AggregateContext<'a>, Row)>> + 'a,
+        labels: Vec<String>,
+        table_alias: &'a str,
+    ) -> Result<impl Stream<Item = Result<Row>> + 'a> {
+        #[derive(futures_enum::Stream)]
+        enum Rows<I1, I2> {
+            NonOrderBy(I1),
+            OrderBy(I2),
+        }
         if self.order_by.is_empty() {
-            let rows = rows.map_ok(|aggregate_context| {
-                let AggregateContext { aggregated, next } = aggregate_context;
+            let rows = rows.map_ok(|(_, row)| row);
 
-                (aggregated.map(Rc::new), next)
-            });
-
-            return Ok(Box::pin(rows));
+            return Ok(Rows::NonOrderBy(Box::pin(rows)));
         }
-
+        let labels = Rc::from(labels.clone());
         let rows = rows
-            .and_then(move |AggregateContext { aggregated, next }| async move {
-                let blend_context = Rc::clone(&next);
+            .and_then(|(AggregateContext { aggregated, next }, row)| {
+                // let table_alias = next.get_table_alias();
+                let labels = Rc::clone(&labels);
                 let filter_context = Rc::new(FilterContext::concat(
                     self.context.as_ref().map(Rc::clone),
                     Some(Rc::clone(&next)),
                 ));
+                let label_context = BlendContext::new(table_alias, labels, Some(row.clone()), None);
```

Comment:
It looks bad to clone `row`. I'm still struggling to fix it.

Reply:
Oh, I think we don't need to make `BlendContext`.
How about making `FilterContext` and chaining it with existing filter_context.
Then we don't need to use `FilterContext::concat`, it can be used just as `next`

Reply:
<img width="290" alt="image" src="https://user-images.githubusercontent.com/2025065/187924047-7d6e185f-07eb-4357-a5ed-4c44c44111c4.png">
We can simply use `FilterContext::new`